crates/da/src/celestia/full_node.rs

```diff
@@ -76,39 +79,65 @@ impl LightDataAvailabilityLayer for CelestiaConnection {
     async fn get_finalized_epoch(&self, height: u64) -> Result<Vec<VerifiableEpoch>> {
         trace!("searching for epoch on da layer at height {}", height);
 
-        match BlobClient::blob_get_all(&self.client, height, &[self.snark_namespace]).await {
-            Ok(maybe_blobs) => match maybe_blobs {
-                Some(blobs) => {
-                    let valid_epochs: Vec<VerifiableEpoch> = blobs
-                        .into_iter()
-                        .filter_map(|blob| {
-                            match FinalizedEpoch::try_from(&blob) {
-                                Ok(epoch) => Some(Box::new(epoch) as VerifiableEpoch),
-                                Err(e) => {
-                                    warn!(
-                                        "Ignoring blob: marshalling blob from height {} to epoch json failed with error {}: {:?}",
-                                        height, e, &blob
-                                    );
-                                    None
-                                }
-                            }
-                        })
-                        .collect();
-                    Ok(valid_epochs)
-                }
-                None => Ok(vec![]),
-            },
-            Err(err) => {
-                if err.to_string().contains("blob: not found") {
-                    Ok(vec![])
-                } else {
-                    Err(anyhow!(DataAvailabilityError::DataRetrievalError(
-                        height,
-                        format!("getting epoch from da layer: {}", err)
-                    )))
+        for attempt in 0..self.fetch_max_retries {
+            match tokio::time::timeout(
```

Comment:
For DRY and to reduce indentation make a method `async fn try_fetch_blobs(height, namespace) -> Result<Blobs>` that internally does the timeout and just returns the blobs, which you can then match after calling it